src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php

```diff
@@ -1061,7 +1061,43 @@ function ($a) {
                                             })
                                         ->end()
                                         ->defaultValue(array())
-                                        ->prototype('scalar')->end()
+                                        ->beforeNormalization()
+                                            ->always()
+                                            ->then(function ($middleware) {
+                                                return array_map(function ($item) {
+                                                    if (\is_string($item)) {
+                                                        return array('id' => $item);
+                                                    }
+
+                                                    return $item;
+                                                }, $middleware);
+                                            })
+                                        ->end()
+                                        ->prototype('array')
+                                            ->beforeNormalization()
+                                                ->always()
+                                                ->then(function (array $middleware) {
+                                                    if (isset($middleware['id'])) {
+                                                        return $middleware;
+                                                    }
+                                                    $arguments = reset($middleware);
+
+                                                    return array(
+                                                        'id' => key($middleware),
```

Comment:
I think we should throw an exception is we have multiple keys in here

Reply:
The issue is that `beforeNormalization` isn't really designed for throwing. It's the `validate` responsibility which would also prepend the path. But at this point we would have already ignored the extra keys.

So we could add manually something like:

```php
if (\count($middleware) > 1) {
    throw new \InvalidArgumentException(sprintf('At path "framework.messenger.buses.middleware": expected a single entry for a middleware item array, with factory id as key and arguments as value. Got "%s".', json_encode($middleware)));
}
```

Do you think it is worth it?

Reply:
I’d say, it’s easy to end up with something like that:
```
middleware:
    - foo: 1
       bar: { baz: ~ }
```